src/common/state/redis.c

```diff
@@ -199,27 +263,105 @@ task *parse_redis_task_table_entry(task_id id,
 void redis_object_table_add_callback(redisAsyncContext *c,
                                      void *r,
                                      void *privdata) {
-  REDIS_CALLBACK_HEADER(db, callback_data, r);
-
-  if (callback_data->done_callback) {
-    task_table_done_callback done_callback = callback_data->done_callback;
-    done_callback(callback_data->id, callback_data->user_context);
+  LOG_DEBUG("Calling object table add callback");
+  REDIS_MULTI_CALLBACK_HEADER(db, callback_data, r, requests_info);
+  redisReply *reply = r;
+  object_id id = callback_data->id;
+  unsigned char *digest = callback_data->data;
+
+#define NUM_CHECK_AND_SET_COMMANDS 3
+#define CHECK_AND_SET_SETNX_INDEX 0
+#define CHECK_AND_SET_GET_INDEX 1
+#define CHECK_AND_SET_SADD_INDEX 2
```

Comment:
Sure, that makes sense. I prefer to keep it in redis.c though, since this is specific to the Redis implementation of the object table.